bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableTableAdminClientWrapper.java

```diff
@@ -306,4 +317,107 @@ public Void apply(Empty input) {
+        GetBackupRequest.newBuilder().setName(clusterName.toBackupName(backupId)).build();
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.getBackupAsync(request),
+        new Function<com.google.bigtable.admin.v2.Backup, Backup>() {
+          @Override
+          public Backup apply(com.google.bigtable.admin.v2.Backup backup) {
+            return Backup.fromProto(backup);
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Backup> updateBackupAsync(UpdateBackupRequest request) {
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.updateBackupAsync(
+            request.toProto(instanceName.getProjectId(), instanceName.getInstanceId())),
+        new Function<com.google.bigtable.admin.v2.Backup, Backup>() {
+          @Override
+          public Backup apply(com.google.bigtable.admin.v2.Backup backup) {
+            return Backup.fromProto(backup);
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<List<String>> listBackupsAsync(String clusterId) {
+    BigtableClusterName clusterName = instanceName.toClusterName(clusterId);
+    ListBackupsRequest request =
+        ListBackupsRequest.newBuilder().setParent(clusterName.getClusterName()).build();
+
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.listBackupsAsync(request),
+        new Function<ListBackupsResponse, List<String>>() {
+          @Override
+          public List<String> apply(ListBackupsResponse response) {
+            List<String> backups = new ArrayList<>();
+            for (com.google.bigtable.admin.v2.Backup backup : response.getBackupsList()) {
+              backups.add(backup.getName());
+            }
+            return backups;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Void> deleteBackupAsync(String clusterId, String backupId) {
+    BigtableClusterName clusterName = instanceName.toClusterName(clusterId);
+    DeleteBackupRequest request =
+        DeleteBackupRequest.newBuilder().setName(clusterName.toBackupName(backupId)).build();
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.deleteBackupAsync(request),
+        new Function<Empty, Void>() {
+          @Override
+          public Void apply(Empty empty) {
+            return null;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<RestoredTableResult> restoreTableAsync(RestoreTableRequest request) {
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.restoreTableAsync(
+            request.toProto(instanceName.getProjectId(), instanceName.getInstanceId())),
+        new Function<Operation, RestoredTableResult>() {
+          @Override
+          public RestoredTableResult apply(Operation operation) {
+            // todo what to put there?
+            try {
+              return new RestoredTableResult(
+                  Table.fromProto(
+                      operation.getResponse().unpack(com.google.bigtable.admin.v2.Table.class)),
+                  "");
```

Comment:
@liubonan, I wasn't sure what to put as the second parameter here (currently `""`), can you advise?

Reply:
I think there was some change between this code was initially written and the java-bigtable repository was created. In the latest PR, restoreTableAsync now returns ApiFuture<RestoredTableResult> instead of an operation.

Reply:
Sorry, by second parameter I meant the String value for this constructor: https://github.com/googleapis/java-bigtable/blob/master/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/RestoredTableResult.java#L33, `String optimizeRestoredTableOperationName`

Reply:
That should be the optimize table operation name from the metadata like this:
`operation.getMetadata().getOptimizeTableOperationName()`.

My concern is the `OptimizeTableOperationName` may not be there when the operation is still running. Therefore, I think restoreTableAsync should return `ApiFuture<RestoredTableResult>` and you need to use token from the result of restoreTableAsync.

Reply:
Ah I see what you're saying. Will fix up. Thanks!

Reply:
discussed offline and fixed up. Resolving